minionsai/discriminator_only/translator.py

```diff
@@ -57,30 +85,122 @@ def translate(self, game: Game):
         opp_money = min(all_money[1 - game.active_player_color], self.MAX_MONEY)
         score_diff = max(min(scores[game.active_player_color] - scores[game.inactive_player_color], self.MAX_SCORE_DIFF), -self.MAX_SCORE_DIFF) + self.MAX_SCORE_DIFF
         
-        return {
-            'board': np.array([board_obs]),  # shape is [batch, num_items=BOARD_SIZE^2]
-            'remaining_turns': np.array([[remaining_turns]]),  # shape is [batch, num_items=1]
-            'money': np.array([[money]]),
-            'opp_money': np.array([[opp_money]]),
-            'score_diff': np.array([[score_diff]])
-        }
+        d = OrderedDict()
+        d['board'] = np.array([board_obs])  # shape is [batch, num_items=BOARD_SIZE^2]
+        d['money'] = np.array([[money]])
+        d['remaining_turns'] = np.array([[remaining_turns]])  # shape is [batch, num_items=1]
+        d['opp_money'] = np.array([[opp_money]])
+        d['score_diff'] = np.array([[score_diff]])
+        if self.mode == "generator":
+            d['phase'] = np.array([[phase]])
+            # legal_actions = [self.valid_actions(game).flatten()]
+            # d['legal_actions'] = np.array([])
+        return d
+
+
+
+    def get_possible_legal_actions(self):
+        possibly_legal = np.zeros((self.num_things, self.num_things), bool)
+        for i in range(BOARD_SIZE ** 2 + 1):
```

Comment:
This loop going to `BOARD_SIZE ** 2 +1` only works if `self.spawn_pos = BOARD_SIZE` I think. It might be simpler to just have the main loop over `BOARD_SIZE ** 2`, and handle spawn separately.

As an example of why it's confusing with the +1 in the loop: if I understand right the entire last loop over `j` is guaranteed to do nothing.